pkg/ottl/ottlfuncs/func_parse_xml.go

```diff
@@ -0,0 +1,134 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package ottlfuncs // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/ottlfuncs"
+
+import (
+	"bytes"
+	"context"
+	"encoding/xml"
+	"errors"
+	"fmt"
+	"strings"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl"
+)
+
+type ParseXMLArguments[K any] struct {
+	Target ottl.StringGetter[K]
+}
+
+func NewParseXMLFactory[K any]() ottl.Factory[K] {
+	return ottl.NewFactory("ParseXML", &ParseXMLArguments[K]{}, createParseXMLFunction[K])
+}
+
+func createParseXMLFunction[K any](_ ottl.FunctionContext, oArgs ottl.Arguments) (ottl.ExprFunc[K], error) {
+	args, ok := oArgs.(*ParseXMLArguments[K])
+
+	if !ok {
+		return nil, fmt.Errorf("ParseXMLFactory args must be of type *ParseXMLArguments[K]")
+	}
+
+	return parseXML(args.Target), nil
+}
+
+// parseXML returns a `pcommon.Map` struct that is a result of parsing the target string as XML
+func parseXML[K any](target ottl.StringGetter[K]) ottl.ExprFunc[K] {
+	return func(ctx context.Context, tCtx K) (any, error) {
+		targetVal, err := target.Get(ctx, tCtx)
+		if err != nil {
+			return nil, err
+		}
+
+		parsedXML := anyXML{}
+
+		decoder := xml.NewDecoder(strings.NewReader(targetVal))
+		err = decoder.Decode(&parsedXML)
+		if err != nil {
+			return nil, fmt.Errorf("unmarshal xml: %w", err)
+		}
+
+		if decoder.InputOffset() != int64(len(targetVal)) {
+			return nil, errors.New("trailing bytes after parsing xml")
+		}
+
+		parsedMap := pcommon.NewMap()
+		parsedXML.intoMap(parsedMap)
+
+		return parsedMap, nil
+	}
+}
+
+type anyXML struct {
```

Comment:
Is this new type a duplicate of what Stanza uses?

Reply:
It is not an exact duplicate, I made a few changes because:
1. the old standalone Stanza seemed to have support for multiple XML elements at the top level (I don't think we want to support this since we don't support multiple rows for csv/key value or multiple json payloads for the other parse converters), and 
2. Stanza had also written the decoding in an iterative way using references back to the element's parent, which I believe was due to 1 here. Doing it recursively helps limit the book keeping which makes it simpler to follow, IMO.

For reference, this is the standalone Stanza implementation, the core of it is largely the same: 
https://github.com/observIQ/stanza/blob/c8336bf1a1f149c4d6f9785c28e8b6fe86e84466/operator/builtin/parser/xml/xml.go#L82

Reply:
Can we pick a different name. `anyXML` doesn't feel right.

Reply:
Sure, do you think something like `xmlNode` or `xmlElement` is more appropriate?

Reply:
xmlElement seems good to me

Reply:
Updated to `xmlElement` 👍